notify/impl.go

```diff
@@ -585,17 +585,27 @@ func (n *Slack) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 	if err != nil {
 		return true, redactURL(err)
 	}
-	resp.Body.Close()
+	defer resp.Body.Close()
 
-	return n.retry(resp.StatusCode)
+	return n.retry(resp)
+}
+
+func slackErr(statusCode int, body io.Reader) error {
+	if body != nil {
+		if bs, err := ioutil.ReadAll(body); err == nil {
+			return fmt.Errorf("unexpected status code %v: %s", statusCode, string(bs))
```

Comment:
Nit but I would do `"unexpected status code %v: %q"` so the returned text is quoted.

Reply:
Done